packages/wrangler/src/pages/build-env.ts

```diff
@@ -34,24 +38,32 @@ export const Handler = async (args: PagesBuildEnvArgs) => {
 	}
 
 	const configPath = path.resolve(args.projectDir, "wrangler.toml");
-	// Fail early if the config file doesn't exist
 	if (!existsSync(configPath)) {
-		throw new FatalError(
-			"No Pages config file found",
-			EXIT_CODE_NO_CONFIG_FOUND
-		);
+		logger.log("No Pages configuration file found. Exiting.");
+		process.exitCode = EXIT_CODE_NO_CONFIG_FOUND;
+		return;
 	}
+
 	logger.log("Reading build configuration from your wrangler.toml file...");
 
-	const config = readConfig(
-		configPath,
-		{
-			...args,
-			// eslint-disable-next-line turbo/no-undeclared-env-vars
-			env: process.env.PAGES_ENVIRONMENT,
-		},
-		true
-	);
+	let config: Omit<Config, "pages_build_output_dir"> & {
+		pages_build_output_dir: string;
+	};
+	try {
+		config = readConfig(
+			configPath,
+			{
+				...args,
+				// eslint-disable-next-line turbo/no-undeclared-env-vars
+				env: process.env.PAGES_ENVIRONMENT,
```

Comment:
Is this a pre-existing thing inside Pages CI or can we formalise into a WRANGLER_ env var with `getEnvironmentVariableFactory`?

Reply:
This is a pre-existing thing in Pages CI